comid/tdx-profile/teeinstanceid.go

```diff
@@ -0,0 +1,177 @@
+// nolint:dupl
+package tdx
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/veraison/corim/encoding"
+)
+
+// TeeInstanceID stores an TEE Instance Identifier. The supported formats are uint and variable-length bytes.
+type TeeInstanceID struct {
+	val interface{}
+}
+
+// NewteeInstanceID creates a new InstanceID from the
+// supplied interface. The supported types are positive integers and
+// byte array
+func NewTeeInstanceID(val interface{}) *TeeInstanceID {
+	switch t := val.(type) {
+	case uint, uint64:
+		return &TeeInstanceID{val: t}
+	case []byte:
+		return &TeeInstanceID{val: t}
+	case int:
+		if t < 0 {
+			return nil
+		}
+		return &TeeInstanceID{val: t}
+	default:
+		return nil
+	}
+}
+
+// SetTeeInstanceID sets the supplied value of Instance ID
+func (o *TeeInstanceID) SetTeeInstanceID(val interface{}) error {
+	switch t := val.(type) {
+	case uint, uint64:
+		o.val = val
+	case []byte:
+		o.val = val
+	case int:
+		if t < 0 {
+			return fmt.Errorf("unsupported negative TeeInstanceID: %d", t)
+		}
+		o.val = val
+	default:
+		return fmt.Errorf("unsupported TeeInstanceID type: %T", t)
+	}
+	return nil
+}
+
+// valid checks for validity of TeeInstanceID and
+// returns an error if Invalid
+func (o TeeInstanceID) Valid() error {
+	if o.val == nil {
+		return fmt.Errorf("empty TeeInstanceID")
+	}
+	switch t := o.val.(type) {
+	case uint, uint64:
+		return nil
+	case []byte:
+		if len(t) == 0 {
+			return fmt.Errorf("empty TeeInstanceID")
+		}
+	case int:
+		if t < 0 {
+			return fmt.Errorf("unsupported negative TeeInstanceID: %d", t)
+		}
+	default:
+		return fmt.Errorf("unsupported TeeInstanceID type: %T", t)
+	}
+	return nil
+}
+
+func (o TeeInstanceID) GetUintTeeInstanceID() (uint, error) {
```

Comment:
Just `GetUint()` would be better, since your calling a method of a `TeeInstanceID` instance, you don't need to repeat the type in the method name.